spec/stud/interval_spec.rb

```diff
@@ -24,6 +35,59 @@
       end
     end
 
+    it 'should be able to interrupt an interval within same thread' do
+      counter = 0
+
+      Stud.interval(0.5) do
+        counter += 1
+        Stud.stop!
+      end
+
+      expect(counter).to eq(1)
+    end
+
+    it 'should be able to interrupt an interval from different threads' do
+      counter = 0
+
+      start_time = Time.now
+      t = Thread.new do
+        Stud.interval(10) do
+          counter += 1
+        end
+      end
+
+      sleep(1)
+      Stud.stop!(t)
+      t.join
+
+      # the execution time worst case should be the sleep(1) call plus a sleep(1) in
+      # the interval loop before seeing the stop? and maybe a "reminder" sleep(< 1)
+      # so < 3 should be safe.
+      expect(Time.now - start_time).to be < 3
+      expect(counter).to eq(1)
+    end
+
+    it 'should be able to interrupt an interval from different threads when sleeping' do
```

Comment:
this "when sleeping" should be a context, where `:sleep_then_run => true` is always set